home/home_2.1/main.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+let FILENAME = "user_stats.txt"
+let DELETE_COMMAND = "delete"
+let BREAD_COMMAND = "bread"
+let EXIT_COMMAND = "exit"
+
+struct StatsManager {
+    var stats: [String: Int] = [:]
+
+    init() {
+        self.stats = readStats()
+    }
+
+    func readStats() -> [String: Int] {
+        var stats = [String: Int]()
+        if let fileContents = try? String(contentsOfFile: FILENAME) {
+            let lines = fileContents.split(separator: "\n")
+            for line in lines {
+                let parts = line.split(separator: " ")
+                if parts.count == 2, let count = Int(parts[1]) {
+                    stats[String(parts[0])] = count
+                }
+            }
+        }
+        return stats
+    }
+
+    func writeStats() {
+        var fileContents = ""
+        for (name, count) in stats {
+            fileContents += "\(name) \(count)\n"
+        }
+        try? fileContents.write(toFile: FILENAME, atomically: true, encoding: .utf8)
+    }
+
+    mutating func deleteStats() {
+        stats.removeAll()
+        try? "".write(toFile: FILENAME, atomically: true, encoding: .utf8)
+    }
+
+    mutating func incrementStat(for name: String) {
+        if let currentCount = stats[name] {
+            stats[name] = currentCount + 1
+        } else {
+            stats[name] = 1
+        }
+        writeStats()
+    }
+
+    mutating func resetStat(for name: String) {
+        stats.removeValue(forKey: name)
+        writeStats()
+    }
+
+    mutating func clearAllStats() {
```

Comment:
this function looks redundant it calls stats.removeAll() and then deleteStats which is also call stats.removeAll() - I hope you should cleanup the code. Also  deleteStats has duplication of writeStats logic. There should be no difference whether you have empty stats or not - writing to file should be the same - it should replace all the content with an actual version of stats serialized to expected format